### PR TITLE
Replace py-h5py~mpi with py-h5py

### PR DESCRIPTION
**Pull request by GianlucaFicarelli**

### CODE REVIEW

**Comment by ferdonline** [comment]
What's the rationale behind this PR? 
AFAIK, the reason for building `~mpi` versions is to enable running programs without an MPI environment (`srun...`), which is required when we link with hpe mpi libraries.

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ class PyMorphio(PythonPackage):
     version('2.1.2', tag='v2.1.2', submodules=True, get_full_repo=True)
     version('2.0.8', tag='v2.0.8', submodules=True, get_full_repo=True)
 
+    variant('mpi', default=True, description="Use parallel HDF5 when needed")
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphio/package.py]: no.

**Review by matz-e — COMMENTED** [review]
@ferdonline wasn't this just a work-around to appease the weird ways we launch stuff on SLURM? CS insisted on having `sbatch` launch a `srun` executing the user script. Libraries that normally don't care then see MPI stuff in the environment and get triggered.

We now switched to `srun --mpi=none` by default, and that should allow us to have the standard behavior of using MPI-enabled libraries in non-MPI contexts. We have a whole bunch of *fake* MPI dependencies just to appease our weird old setup.

@GianlucaFicarelli I'd rather remove `morphio±mpi` specs…

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 1 above)
Comment (GianlucaFicarelli) [reply var/spack/repos/builtin/packages/py-morphio/package.py]: Thanks, you're right, I was holding the fix to allow the previous jenkins build to finish.

File: var/spack/repos/builtin/packages/brion/package.py [hunk 2]
```diff
@@ -36,17 +36,17 @@ class Brion(CMakePackage):
...
     # issue (libboost_iostreams.so.1.68.0 not finding libbz2.so)
     depends_on('bzip2')
     depends_on('lunchbox', when='@3.1.0')
     depends_on('vmmlib', when='@3.1.0')
-    depends_on('highfive +boost ~mpi')
-    depends_on('highfive@2.1.1 +boost ~mpi', when='@3.1.0')
-    depends_on('mvdtool ~mpi')
+    depends_on('highfive +boost')
+    depends_on('highfive@2.1.1 +boost', when='@3.1.0')
+    depends_on('mvdtool')
```
Comment (GianlucaFicarelli) [inline var/spack/repos/builtin/packages/brion/package.py:49]: I'm not sure that removing `~mpi` in this file the right thing to do, but without this the concretization was failing. Any advice?

**Comment by pramodk** [comment]
Retest this please

**Comment by pramodk** [comment]
> @ferdonline wasn't this just a work-around to appease the weird ways we launch stuff on SLURM? CS insisted on having sbatch launch a srun executing the user script. Libraries that normally don't care then see MPI stuff in the environment and get triggered.

@ferdonline : we had issue because of old HPE-MPI and SLURM settings on BB5 which got resolved via this ticket [HELP-12377](https://bbpteam.epfl.ch/project/issues/browse/HELP-12377).

We accumulated this `+mpi / ~mpi` over different packages and it's creating concretisation issues. In Spack, `+mpi` is default and hence we are trying to the same defaults everywhere consistently.

File: var/spack/repos/builtin/packages/brion/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/brion/package.py:49]: I think that it's best to be consistent. File seems good as it is, we should rather educate users on doing the right thing when launching their software.

**Review by matz-e — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-morphio/package.py [hunk 3]
```diff
@@ -22,12 +22,8 @@ class PyMorphio(PythonPackage):
...
 
-    variant('mpi', default=False, description="Use parallel HDF5 when needed")
-
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm', type='build')
 
     depends_on('cmake@3.2:', type='build')
     depends_on('py-numpy', type='run')
-    depends_on('hdf5 +mpi', type=('build', 'run'), when='+mpi')
-    depends_on('hdf5 ~mpi', type=('build', 'run'), when='~mpi')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-morphio/package.py:33]: @GianlucaFicarelli : doesn't py-morphio required hdf5 dependency? (or highfive?)

**Review by pramodk — COMMENTED** [review]
LGTM.

I will let @matz-e to verify it :)

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 3 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-morphio/package.py:33]: Yes, that should stay in here as just `depends_on('hdf5')`. `highfive` comes as a submodule, @pramodk .

**Review by matz-e — CHANGES_REQUESTED** [review]
We're getting very close!

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 3 above)
Comment (GianlucaFicarelli) [reply var/spack/repos/builtin/packages/py-morphio/package.py:33]: Yes, I wanted to rollback https://github.com/BlueBrain/spack/pull/1081 but I should leave the `hdf5` dependency without specifying `mpi`.

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-morphio/package.py (hunk 3 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/py-morphio/package.py:33]: yes! if you fix this, we should be good to go.

**Comment by ferdonline** [comment]
> We now switched to `srun --mpi=none `

Nice!

**Pull request merged**